### ISSUE TITLE
Allow configuration to force processing of rules on methods marked @Suppress or @SuppressWarning

### ISSUE THREAD

**Issue by JeffHoover** [detekt/detekt#3749]
## Expected Behavior
If `ignoreSupress: true` is in the `config:` section of the configuration file, _process rules on all methods_, even methods marked with @ Suppress

if `ignoreSupressWarning: true` is in the `config:` section of the configuration file, _process rules on all methods_, even methods marked with @ SuppressWarning

## Current Behavior
(As designed) when a method is marked with @ Suppress or @ SuppressWarning, the specified rule is not reported for that method

## Context
I've been asked to analyze code that I don't own. I'd like a complete picture of results, not just the results that the author(s) want to see.

**Comment by BraisGabin** [comment]
I like this idea, it have sense. Even for your own code you may want to revisit all those supresed warnings from time to time.

I think that it should be only one configuration instead of two and it should disable all the suppression: annotations and baseline too.

**Comment by cortinico** [comment]
Agree, that's really valuable to track.
I was also thinking that right now we just "discard" a report if a `@Suppress` is discovered. Instead we could carry over this information on the report, and display a note like "X issues were identified but not reported as they were suppressed in the baseline/locally, etc."

**Comment by BraisGabin** [comment]
Good idea! Probably that's a big breaking change but we should do that for 2.0. I imagine the API like this:

```kotlin
class Finding(
  ...
  val suppressedReasons: List<String>
) {
  val suppressed: Boolean
    get() = suppressedReasons.isNotEmpty()
}
```

**Comment by BraisGabin** [comment]
I was working on this issue and was thinking what to do with the suppressors. Even with that name we should probably don't allow to inspection them. If I configure a rule to don't check functions with `@Composable` annotation we shouldn't consider that like a "suppresion", it's part of the rule configuration. It's similar to say "don't complain about the name of overrided functions". We could consider that a kind of "suppression" but it have really little sense.

**Comment by BraisGabin** [comment]
This is done already. The reports get all the issues found by detekt. Then, depending on the report, it filter out the ones suppressed or not.
